app/controllers/application_controller.rb

```diff
@@ -2,4 +2,14 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user
+
+  def current_user
+    #look up once
+    @current_user ||= User.find(session[:uid]) if session[:uid]
```

Comment:
This method was actually ok before. The only thing that needed to change was line 4 of SessionsController. The key in the session here, should match the key in the session on line 4 in the SessionsController.